src/Semantic/Api/Symbols.hs

```diff
@@ -90,38 +92,15 @@ tagToSymbol Tag{..} = Symbol
   }
 
 
-data ALaCarteTerm syntax ann = ALaCarteTerm Language [Text] (Term syntax ann)
-
-instance IsTaggable syntax => Precise.ToTags (ALaCarteTerm syntax) where
-  tags source (ALaCarteTerm lang symbolsToSummarize term) = runTagging lang source symbolsToSummarize term
-
-
-doParse
-  :: ( Carrier sig m
-     , Member (Error SomeException) sig
-     , Member Parse sig
-     , Member (Reader PerLanguageModes) sig
-     )
-  => (forall t . Precise.ToTags t => t Loc -> m a)
-  -> [Text]
-  -> Blob
-  -> m a
-doParse with symbolsToSummarize blob = do
-  modes <- ask @PerLanguageModes
-  case blobLanguage blob of
-    Go         -> parse Parser.goParser         blob >>= with . mkTerm
-    Haskell    -> parse Parser.haskellParser    blob >>= with . mkTerm
-    JavaScript -> parse Parser.tsxParser        blob >>= with . mkTerm
-    JSON       -> parse Parser.jsonParser       blob >>= with . mkTerm
-    JSX        -> parse Parser.tsxParser        blob >>= with . mkTerm
-    Markdown   -> parse Parser.markdownParser   blob >>= with . mkTerm
-    Python
-      | Precise <- pythonMode modes -> parse Parser.precisePythonParser blob >>= with
-      | otherwise                   -> parse Parser.pythonParser        blob >>= with . mkTerm
-    Ruby       -> parse Parser.rubyParser       blob >>= with . mkTerm
-    TypeScript -> parse Parser.typescriptParser blob >>= with . mkTerm
-    TSX        -> parse Parser.tsxParser        blob >>= with . mkTerm
-    PHP        -> parse Parser.phpParser        blob >>= with . mkTerm
-    _          -> noLanguageForBlob (blobPath blob)
-    where mkTerm :: Term syntax Loc -> ALaCarteTerm syntax Loc
-          mkTerm = ALaCarteTerm (blobLanguage blob) symbolsToSummarize
+class ToTags t where
+  tags :: Language -> [Text] -> Source -> t Loc -> [Tag]
```

Comment:
`ALaCarteTerm` prevented us being able to use the new parser abstraction facilities, so I’ve instead _further_ abstracted tag generation behind this local class, implemented using `runTagging` on the à la carte side and `Precise.ToTags` on the precise side.